spec/unit/pops/types/p_object_type_spec.rb

```diff
@@ -0,0 +1,599 @@
+require 'spec_helper'
+require 'puppet/pops'
+
+module Puppet::Pops
+module Types
+describe 'The Object Type' do
+  let(:parser) { TypeParser.new }
+  let(:pp_parser) { Puppet::Pops::Parser::EvaluatingParser.new }
+  let(:loader) { Puppet::Pops::Loader::BaseLoader.new(nil, 'type_parser_unit_test_loader') }
+
+  def type_object_t(name, body_string)
+    TypeFactory.type_alias(name, pp_parser.parse_string("Object[{#{body_string}}]").current)
+  end
+
+  def expect_object(name, body_string)
+    obj = type_object_t(name, body_string)
+    loader.expects(:load).with(:type, name.downcase).at_least_once.returns obj
+    obj
+  end
+
+  def parse_object(name, body_string)
+    expect_object(name, body_string)
+    parser.parse(name, loader)
+  end
+
+  context 'when dealing with attributes' do
+    it 'raises an error when the attribute type is not a type' do
+      obj = <<-OBJECT
+        attributes => {
+          a => 23
+        }
+      OBJECT
+      expect { parse_object('MyObject', obj) }.to raise_error(TypeAssertionError,
+        /attribute 'a' had wrong type, expected a Type value, got Integer/)
```

Comment:
the past tense throws me every time  when it is reporting this error 'a' still has the wrong type unless it was magically repaired